packages/grafana-ui/src/components/GraphNG/nullToValue.ts

```diff
@@ -1,17 +1,25 @@
-import { DataFrame } from '@grafana/data';
+import { ArrayVector, DataFrame } from '@grafana/data';
 
 export function nullToValue(frame: DataFrame) {
-  frame.fields.forEach((f) => {
-    const noValue = +f.config?.noValue!;
-    if (!Number.isNaN(noValue)) {
-      const values = f.values.toArray();
-      for (let i = 0; i < values.length; i++) {
-        if (values[i] === null) {
-          values[i] = noValue;
+  return {
+    ...frame,
+    fields: frame.fields.map((field, i) => {
+      const noValue = +field.config?.noValue!;
+      const values = field.values.toArray().slice();
+      const transformedVals: any[] = [];
+
+      if (!Number.isNaN(noValue)) {
+        for (let i = 0; i < values.length; i++) {
+          transformedVals[i] = values[i] === null ? noValue : values[i];
```

Comment:
```suggestion
      if (!Number.isNaN(noValue)) {
        const transformedVals = field.values.toArray().slice();

        for (let i = 0; i < transformedVals.length; i++) {
          if (transformedVals[i] === null) {
            transformedVals[i] = noValue;
          }
```

Reply:
Ah OK, that makes sense. Definitely cleaner than what I had done.